### PR TITLE
[15.0][ADD] auth_signup_partner_company

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
@qrtl

### CODE REVIEW

File: auth_signup_partner_company/models/res_users.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+            .sudo()
+            .get_param("base.template_portal_user_id", "False")
+        )
+        template_user = self.browse(template_user_id)
+        if not (self.env.company.id in template_user.company_ids.ids):
+            template_user.company_ids = [(4, self.env.company.id)]
+        user = super()._signup_create_user(values)
+        user.company_id = self.env.company
+        user.partner_id.company_id = self.env.company
+        return user
```
Comment (yostashiro) [inline auth_signup_partner_company/models/res_users.py]:
```suggestion
    def _create_user_from_template(self, values):
        values["company_ids"] = [self.env.company.id]
        values["company_id"] = self.env.company.id
        return super()._create_user_from_template(values)

    def _signup_create_user(self, values):
        user = super()._signup_create_user(values)
        user.partner_id.company_id = self.env.company
        return user
```
Can you try and see if this works fine.

File: auth_signup_partner_company/readme/DESCRIPTION.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+This module assigns company to the newly created user and its partner.
```
Comment (yostashiro) [inline auth_signup_partner_company/readme/DESCRIPTION.rst]:
```suggestion
This module assigns company to the user and its partner newly created through portal signup.

Odoo's standard behavior when a new portal user signs up:

* User's company is taken from the template user.
* Partner which links to the new user does not have a company assigned.

This behavior is problematic under multi-company, multi-website settings where partners
should be seggregated between the companies, and this module intends to fix this issue.
```

**Review by yostashiro — CHANGES_REQUESTED** [review]
Can you please add some tests.

**Comment by kanda999** [comment]
Function review: it worked as expected.

**Review by kanda999 — APPROVED** [review]

File: auth_signup_partner_company/tests/test_auth_signup_partner_company.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        super(TestAuthSignupPartnerCompany, self).setUp()
+        self.env["ir.config_parameter"].set_param("auth_signup.invitation_scope", "b2c")
+        self.company = self.env.company
+        self.company_jpy = self.env["res.company"].create(
+            {
+                "name": "Japan company",
+                "currency_id": self.env.ref("base.JPY").id,
+                "country_id": self.env.ref("base.jp").id,
+            }
+        )
```
Comment (yostashiro) [inline auth_signup_partner_company/tests/test_auth_signup_partner_company.py]:
```suggestion
        self.company_1 = self.env.company
        self.company_2 = self.env["res.company"].create({"name": "company 2"})
```

Let's remove the attributes not used in the test.

File: auth_signup_partner_company/tests/test_auth_signup_partner_company.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+        self.assertEqual(user.company_id, self.company)
+        self.assertEqual(user.partner_id.company_id, self.company)
+
+        self.env.company = self.company_jpy
+        self.env["res.users"].signup(
+            {"name": "Test1", "login": "test1@gmail.com", "password": "12345678"}
+        )
+        user = self.env["res.users"].search([("login", "=", "test1@gmail.com")])
+        self.assertEqual(user.company_id, self.company_jpy)
+        self.assertEqual(user.partner_id.company_id, self.company_jpy)
```
Comment (yostashiro) [inline auth_signup_partner_company/tests/test_auth_signup_partner_company.py]:
```suggestion
    def test_auth_signup_company(self):
        self.assertEqual(self.env.company, self.company_1)
        self.env["res.users"].signup(
            {"name": "test 1", "login": "test1@gmail.com", "password": "123"}
        )
        user = self.env["res.users"].search([("login", "=", "test1@gmail.com")])
        self.assertEqual(user.company_id, self.company_1)
        self.assertEqual(user.partner_id.company_id, self.company_1)

        # Switch to another company and sign up.
        self.env.company = self.company_2
        self.env["res.users"].signup(
            {"name": "test 2", "login": "test2@gmail.com", "password": "123"}
        )
        user = self.env["res.users"].search([("login", "=", "test2@gmail.com")])
        self.assertEqual(user.company_id, self.company_2)
        self.assertEqual(user.partner_id.company_id, self.company_2)
```

**Review by yostashiro — COMMENTED** [review]
Just from the preference of not adding unnecessary contexts to the test data.

**Review by yostashiro — APPROVED** [review]
Code review. LGTM. 👍

**Pull request merged**